store/src/java-test/com/zimbra/cs/service/formatter/MobileConfigFormatterTest.java

```diff
@@ -0,0 +1,273 @@
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.cs.service.formatter;
+
+import java.io.StringWriter;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+import com.google.common.collect.Maps;
+import com.zimbra.common.mime.MimeConstants;
+import com.zimbra.cs.account.Account;
+import com.zimbra.cs.account.Domain;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.account.Server;
+import com.zimbra.cs.mailbox.MailboxTestUtil;
+import com.zimbra.cs.service.formatter.FormatterFactory.FormatType;
+import com.zimbra.cs.service.formatter.MobileConfigFormatter.ConfigType;
+
+public class MobileConfigFormatterTest {
+    private static Formatter formatter;
+    private static final String USER_NAME = "test1@test.domain.com";
+    private static final String DOMAIN_NAME = "test.domain.com";
+
+    @BeforeClass
+    public static void init() throws Exception {
+        MailboxTestUtil.initServer("store/");
+        Provisioning prov = Provisioning.getInstance();
+        prov.createDomain(DOMAIN_NAME, Maps.<String, Object>newHashMap());
+        prov.createAccount(USER_NAME, "secret", Maps.<String, Object>newHashMap());
+        formatter = new MobileConfigFormatter();
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        MailboxTestUtil.clearData("store/");
+    }
+
+    @Test
+    public void testGetType() throws Exception {
+        Assert.assertEquals(FormatType.MOBILE_CONFIG, formatter.getType());
+    }
+
+    @Test
+    public void testGetDefaultMimeTypes() throws Exception {
+        String[] expectedArray = new String[] { MimeConstants.CT_TEXT_XML, "text/xml" };
+        String[] actualArray = formatter.getDefaultMimeTypes();
+        Assert.assertEquals(2, actualArray.length);
+        Assert.assertArrayEquals(expectedArray, actualArray);
+    }
+
+    @Test
+    public void testGetDictForCaldav() throws Exception {
```

Comment:
the below 3 tests have lots of common code duplicated in each test. should refactor.

Reply:
done